apisix/init.lua

```diff
@@ -629,6 +630,12 @@ end
 function _M.http_header_filter_phase()
     core.response.set_header("Server", ver_header)
 
+    local status_code = tonumber(get_var("upstream_status"))
+    if status_code >= 500 and status_code <=599 then
```

Comment:
Not all requests will be proxied to upstream, requests terminated by APISIX itself will have null value of `$upstream_status`, we have to check the validity of `status_code`.

Reply:
BTW, the corresponding test cases should be replenished.

Reply:
Good suggestion, updated.